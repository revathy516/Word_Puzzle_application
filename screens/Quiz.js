import React, { useState } from 'react';
import { View, Button, StyleSheet } from 'react-native';
import questions from '../assets/questions.json';
import QuesAnsPair from '../components/QuesAnsPair';
import { writeScore } from '../assets/scoreStorage';


const Quiz = (props) => {
    const [questionIndex, setQuestionIndex] = useState(0);
    const [showNext, setShowNext] = useState(false);
    const [score, setScore] = useState(0);

    const handleQuizTraversal = () => {
        if (questionIndex === questions['questions'].length - 1) {
            writeScore(`${score}`);
            props.navigation.navigate('LeaderBoard', {score: score});
            return
            //${score} out of ${questions['questions'].length}`
        }
        setQuestionIndex((questionIndex) => questionIndex + 1);
        setShowNext(false);
    }

    const is_next = () => {
        setShowNext(true);
    }

    const get_Score = (score) => {
        setScore(score);
    }


    return (
        <View style={styles.quizContainer}>
            <QuesAnsPair
                question={questions['questions'][questionIndex]['questionText']}
                index={questionIndex}
                answers={questions['questions'][questionIndex]['answers']}
                is_next={is_next}
                getScore={get_Score}
                length={questions['questions'].length}    
            />
                        <Button title={questionIndex === questions['questions'].length - 1 ? 'end ' : 'next'} onPress={handleQuizTraversal} />
        </View>
    );
}

const styles = StyleSheet.create({
    buttonContainer: {
        justifyContent: "space-between",
    },
    backButton: {
        marginRight: 10,
    },
    quizContainer: {
        margin: 20,
        padding: 15,
        minHeight:90
    },

});

export default Quiz;