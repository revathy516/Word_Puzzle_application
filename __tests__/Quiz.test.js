import React from 'react';
import renderer from 'react-test-renderer';
import Quiz from '../screens/Quiz';

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

it('renders correctly', () => {
    const tree = renderer.create(<Quiz />).toJSON();
    expect(tree).toMatchSnapshot();
  });