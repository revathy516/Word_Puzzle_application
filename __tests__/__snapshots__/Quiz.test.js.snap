// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "margin": 20,
      "minHeight": 90,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#33DAFF",
        "borderColor": "#C2185B",
        "borderRadius": 2,
        "borderWidth": 2,
        "margin": 10,
        "minHeight": "20%",
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#C2185B",
          "fontSize": 22,
          "fontWeight": "bold",
          "marginBottom": 15,
          "textAlign": "center",
        }
      }
    >
      Question 
      1
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      Capital of Singapore
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 10,
        "minHeight": "50%",
      }
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "flex": 1,
          "justifyContent": "space-between",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "#d3d6da",
                  "borderWidth": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 5,
                  "width": 40,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#878a8c",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "#d3d6da",
                  "borderWidth": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 5,
                  "width": 40,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#878a8c",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "#d3d6da",
                  "borderWidth": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 5,
                  "width": 40,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#878a8c",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "#d3d6da",
                  "borderWidth": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 5,
                  "width": 40,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#878a8c",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "#d3d6da",
                  "borderWidth": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 5,
                  "width": 40,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#878a8c",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "#d3d6da",
                  "borderWidth": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 5,
                  "width": 40,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#878a8c",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "#d3d6da",
                  "borderWidth": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 5,
                  "width": 40,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#878a8c",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "#d3d6da",
                  "borderWidth": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 5,
                  "width": 40,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#878a8c",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "#d3d6da",
                  "borderWidth": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 5,
                  "width": 40,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#878a8c",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "#d3d6da",
                  "borderWidth": 2,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 5,
                  "width": 40,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#878a8c",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 10,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#d3d6da",
                      "borderRadius": 5,
                      "margin": 3,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                        "fontWeight": "500",
                      }
                    }
                  >
                    r
                  </Text>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#d3d6da",
                      "borderRadius": 5,
                      "margin": 3,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                        "fontWeight": "500",
                      }
                    }
                  >
                    n
                  </Text>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#d3d6da",
                      "borderRadius": 5,
                      "margin": 3,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                        "fontWeight": "500",
                      }
                    }
                  >
                    o
                  </Text>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#d3d6da",
                      "borderRadius": 5,
                      "margin": 3,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                        "fontWeight": "500",
                      }
                    }
                  >
                    p
                  </Text>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#d3d6da",
                      "borderRadius": 5,
                      "margin": 3,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                        "fontWeight": "500",
                      }
                    }
                  >
                    g
                  </Text>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#d3d6da",
                      "borderRadius": 5,
                      "margin": 3,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                        "fontWeight": "500",
                      }
                    }
                  >
                    a
                  </Text>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#d3d6da",
                      "borderRadius": 5,
                      "margin": 3,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                        "fontWeight": "500",
                      }
                    }
                  >
                    e
                  </Text>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#d3d6da",
                      "borderRadius": 5,
                      "margin": 3,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                        "fontWeight": "500",
                      }
                    }
                  >
                    i
                  </Text>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#d3d6da",
                      "borderRadius": 5,
                      "margin": 3,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                        "fontWeight": "500",
                      }
                    }
                  >
                    s
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 10,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#d3d6da",
                      "borderRadius": 5,
                      "margin": 3,
                      "padding": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 15,
                        "fontWeight": "500",
                      }
                    }
                  >
                    ⌫
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 10,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#d3d6da",
                    "borderRadius": 5,
                    "margin": 3,
                    "padding": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 15,
                      "fontWeight": "500",
                    }
                  }
                >
                  ENTER
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        next
      </Text>
    </View>
  </View>
</View>
`;
